pandas/core/categorical.py

```diff
@@ -1753,6 +1753,11 @@ def codes(self):
         from pandas import Series
         return Series(self.categorical.codes, index=self.index)
 
+    @property
+    def density(self):
+        """The cardinality ratio of this categorical."""
+        return np.divide(float(len(self.categorical.categories)), len(self.categorical))
```

Comment:
What would be a more reasonable return value in this instance? `nan`? I think an `inf` value suggests that `k >> n` which could be useful (ie. categories are defined but the categorical is empty) as opposed to `nan` which implies that ratio is meaningless: `k = n = 0`.